addons/show-bbcode/userscript.js

```diff
@@ -0,0 +1,58 @@
+function viewSource(post, msg) {
+  return function (event) {
+    event.preventDefault();
+    const body = post.querySelector(".postmsg");
+    if (event.target.getAttribute("data-state") === "post") {
+      event.target.innerText = msg("source-button-active");
+      event.target.removeAttribute("title");
+      if (event.target.originalHTML === undefined) {
+        event.target.originalHTML = body.firstElementChild;
+      }
+      body.removeChild(body.firstElementChild);
+      const source = document.createElement("div");
+      body.insertBefore(source, body.firstElementChild);
+      source.className = "post_body_html";
+      if (event.target.sourceText !== undefined) {
+        event.target.setAttribute("data-state", "source");
+        source.innerText = event.target.sourceText;
+        return;
+      }
+      event.target.setAttribute("data-state", "loading");
+      source.innerText = msg("loading");
+      fetch("https://scratch.mit.edu/discuss/post/" + post.id.substring(1) + "/source/").then(function (res) {
+        res.text().then(function (text) {
+          event.target.setAttribute("data-state", "source");
+          source.innerText = event.target.sourceText = text;
+        });
+      });
+    } else if (event.target.getAttribute("data-state") === "source") {
+      event.target.innerText = msg("source-button");
+      event.target.title = msg("source-button-tooltip");
+      event.target.setAttribute("data-state", "post");
+      body.removeChild(body.firstElementChild);
+      body.insertBefore(event.target.originalHTML, body.firstElementChild);
+    }
+  };
+}
+
+export default async function ({ addon, console, msg }) {
+  for (const post of document.querySelectorAll(".blockpost")) {
```

Comment:
You should use while (true) {const post = await addon.tab.waitForElement(".blockpost")...} like in better-quoter so that when load more is enabled the newly loaded posts get it added as well

Reply:
Thanks, I'll change that.

Reply:
Infinite scrolling doesn't currently support topics, so this shouldn't be a problem, but I changed it anyway.